Tests/CatalogItems/Write-RsFolderContent.Tests.ps1

```diff
@@ -0,0 +1,90 @@
+        Write-RsFolderContent -Path $localReportPath -RsFolder $folderPath -ReportServerUri $reportServerUri
+        $uploadedReport = (Get-RsFolderContent -RsFolder $folderPath -Recurse ) | Where-Object TypeName -eq 'Report'
+        It "Should upload a local report in Report Server with ReportServerUri Parameter" {
+            $uploadedReport.TypeName | Should Be 'Report'
+        }
+        # Removing folders used for testing
+        Remove-RsCatalogItem -RsFolder $folderPath
+    }
+
+    Context "Write-RsFolderContent with Proxy Parameter"{
+        $folderName = 'SutWriteRsFolderContentProxy' + [guid]::NewGuid()
+        New-RsFolder -Path / -FolderName $folderName
+        $folderPath = '/' + $folderName
+        $localReportPath =   (Get-Item -Path ".\" -Verbose).FullName  + '\Tests\CatalogItems\testResources'
+        $proxy = New-RsWebServiceProxy 
+        Write-RsFolderContent -Path $localReportPath -RsFolder $folderPath -Proxy $proxy
+        $uploadedReport = (Get-RsFolderContent -RsFolder $folderPath -Recurse ) | Where-Object TypeName -eq 'Report'
+        It "Should upload a local report in Report Server with Proxy Parameter" {
+            $uploadedReport.TypeName | Should Be 'Report'
+        }
+        # Removing folders used for testing
+        Remove-RsCatalogItem -RsFolder $folderPath
+    }
+
+    Context "Write-RsFolderContent with Proxy and ReportServerUri"{
+        $folderName = 'SutWriteRsFolderContentAll' + [guid]::NewGuid()
+        New-RsFolder -Path / -FolderName $folderName
+        $folderPath = '/' + $folderName
+        $localReportPath =   (Get-Item -Path ".\" -Verbose).FullName  + '\Tests\CatalogItems\testResources'
+        $proxy = New-RsWebServiceProxy 
+        $reportServerUri = 'http://localhost/reportserver'
+        Write-RsFolderContent -Path $localReportPath -RsFolder $folderPath -Proxy $proxy -ReportServerUri $reportServerUri
+        $uploadedReport = (Get-RsFolderContent -RsFolder $folderPath -Recurse ) | Where-Object TypeName -eq 'Report'
+        It "Should upload a local report in Report Server with ReportServerUri and Proxy Parameters" {
+            $uploadedReport.TypeName | Should Be 'Report'
+        }
+        # Removing folders used for testing
+        Remove-RsCatalogItem -RsFolder $folderPath
+    }
+
+     Context "Write-RsFolderContent with Recurse Parameter"{
+        $folderName = 'SutWriteRsFolderContentRecurse' + [guid]::NewGuid()
+        New-RsFolder -Path / -FolderName $folderName
+        $folderPath = '/' + $folderName
+        $localReportPath =   (Get-Item -Path ".\" -Verbose).FullName  + '\Tests\CatalogItems\testResources'
+        Write-RsFolderContent -Path $localReportPath -RsFolder $folderPath -Recurse
+        $uploadedReports = (Get-RsFolderContent -RsFolder $folderPath -Recurse ) | Where-Object TypeName -eq 'Report'
+        It "Should upload a report that is in a folder and a second report in a subfolder" {
+            $uploadedReports.Count | Should Be 2
```

Comment:
Please also validate all the content you have in the test resources is available in the server after running the cmdlet every folder every dataset and every datasource